### PR TITLE
Feat/build project structure

### PR DESCRIPTION
**Pull request by NhatNguyen1502**

### CODE REVIEW

File: src/db/data-source.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import 'reflect-metadata';
+import { DataSource, DataSourceOptions } from 'typeorm';
+
+export const dataSourceOptions: DataSourceOptions = {
+  type: 'mysql',
+  host: process.env.DB_HOST,
+  port: Number(process.env.DB_PORT),
+  username: process.env.DB_USER || 'root',
```
Comment (vanglevan) [inline src/db/data-source.ts:8]: It's better if you create a constant for default values of these: 'root', 'mysql3306', 'open_universe'

File: src/post/post.service.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { InjectModel } from '@nestjs/mongoose';
+import { Post } from './schemas/post.schema';
+import { Model } from 'mongoose';
+
+@Injectable()
+export class PostService {
+  constructor(@InjectModel(Post.name) private postModel: Model<Post>) {}
+
+  async create(createPostDto: CreatePostDto): Promise<Post> {
+    console.log(createPostDto);
```
Comment (vanglevan) [inline src/post/post.service.ts]: remove the log in source code

File: src/user/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { Controller, Get, Post, Body, Patch, Param, Delete } from '@nestjs/common';
+import { UserService } from './user.service';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+
+@Controller('user')
```
Comment (vanglevan) [inline src/user/user.controller.ts]: The resource must be a plural nouns.
Suggestion: user => users

File: src/user/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class UserService {
+
+  constructor(@InjectRepository(User) private userRepository: Repository<User>) {}
+  async create(createUserDto: CreateUserDto): Promise<User> {
+    const {user_name, email, password} = createUserDto;
+    const newUser = new User();
+      newUser.user_name = user_name;
```
Comment (vanglevan) [inline src/user/user.service.ts]: Is the indent wrong here?

File: src/db/migrations/1712421670682-create-user-table.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class CreateUserTable1712421670682 implements MigrationInterface {
+    name = 'CreateUserTable1712421670682'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`CREATE TABLE \`user\` (\`id\` int NOT NULL AUTO_INCREMENT, \`full_name\` varchar(255) NOT NULL, \`user_name\` varchar(255) NOT NULL, \`date_of_birth\` date NULL, \`password\` varchar(100) NOT NULL, \`email\` varchar(255) NOT NULL, \`phone_number\` varchar(12) NOT NULL, \`role\` enum ('admin', 'user') NOT NULL DEFAULT 'user', \`gender\` enum ('male', 'female') NOT NULL DEFAULT 'male', \`status\` enum ('active', 'inactive') NOT NULL DEFAULT 'active', \`create_at\` timestamp(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6), \`update_at\` timestamp(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6), PRIMARY KEY (\`id\`)) ENGINE=InnoDB`);
```
Comment (vanglevan) [inline src/db/migrations/1712421670682-create-user-table.ts]: Do you have any idea to re-format this query?

File: src/post/dto/create-post.dto.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+export class CreatePostDto {
+  readonly user_id;
```
Comment (vanglevan) [inline src/post/dto/create-post.dto.ts]: Normally, we will follow the camelCase rule in Javascript/Typescript. Do you have a other reason to use under_score rule here?

File: src/user/entities/user.entity.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+  @Column({ type: 'enum', enum: ActiveStatus, default: ActiveStatus.ACTIVE })
+  status: ActiveStatus;
+
+  @CreateDateColumn({ type: 'timestamp' })
+  create_at: Timestamp;
+
+  @UpdateDateColumn({ type: 'timestamp' })
+  update_at: Timestamp;
+}
```
Comment (vanglevan) [inline src/user/entities/user.entity.ts:41]: I think we need to add properties: create_by, update_by to entity.
Please update your database schema and also update other entity if needed

File: .env-example [hunk 8]
```diff
@@ -0,0 +1,6 @@
+PORT=
+DB_HOST=
```
Comment (vanglevan) [inline .env-example:2]: test review

File: .env-example [hunk 9]
```diff
@@ -0,0 +1,6 @@
...
+DB_PORT=
+DB_USER=
+DB_PASSWORD=
```
Comment (vanglevan) [inline .env-example:5]: test

**Review by vanglevan — COMMENTED** [review]
Finish review

File: src/db/migrations/1712421670682-create-user-table.ts (hunk 5 above)
Comment (NhatNguyen1502) [reply src/db/migrations/1712421670682-create-user-table.ts]: How about wrap word?

**Review by vanglevan — APPROVED** [review]

**Pull request merged**